pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -918,6 +926,35 @@ func (ctrl *PersistentVolumeController) updateVolumePhaseWithEvent(volume *v1.Pe
 	return newVol, nil
 }
 
+// assignDefaultStorageClass looks up all storage classes and attempts to assign a storage class to the given claim.
+// The storage class assigned is always the default one and if there is none or there are more defaults the claim
+// will not get updated.
+func (ctrl *PersistentVolumeController) assignDefaultStorageClass(claim *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	// [Unit test set 15]
+	if claim.Spec.StorageClassName != nil {
+		klog.V(4).Infof("can not assign storage class to PersistentVolumeClaim[%s]: pvc already has StorageClass[%s]", claimToClaimKey(claim), *claim.Spec.StorageClassName)
+		return claim, nil
+	}
+
+	class, err := util.GetDefaultClass(ctrl.classLister)
+	if err != nil {
+		klog.V(4).Infof("failed to get default storage class: %v", err)
+		return claim, nil
```

Comment:
Why are you not returning the error?

Should this be `Errorf()` ?

Reply:
`GetDefaultClass` returns errors either when it fails to list SCs or when there is more than one default. 

Anyway, it should be safe to ignore these errors - adding a comment and a link to PR where the function is improved.